app/kubernetes-ingress-controller/gateway-api.md

```diff
@@ -2,19 +2,179 @@
+apiVersion: gateway.networking.k8s.io/v1
+kind: Gateway
+metadata:
+  name: example
+spec:
+  gatewayClassName: kong
+  listeners:
+  - name: proxy
+    port: 80
+    protocol: HTTP
+  - name: proxy-ssl
+    port: 443
+    protocol: HTTPS
+    hostname: kong.example.com
+    tls:
+      mode: Terminate
+      certificateRefs:
+        - kind: Secret
+          name: kong-example-com-cert
+  - name: proxy-tcp-9901
+    port: 9901
+    protocol: TCP
+  - name: proxy-udp-9902
+    port: 9902
+    protocol: UDP
+  - name: proxy-tls-9903
+    port: 9903
+    protocol: TLS
+```
+
+It requires a proxy Service that includes all the requested listener ports:
+
+```yaml
+apiVersion: v1
+kind: Service
+metadata:
+  name: proxy
+spec:
+  ports:
+  - port: 80
+    protocol: TCP
+    targetPort: 8000
+  - port: 443
+    protocol: TCP
+    targetPort: 8443
+  - port: 9901
+    protocol: TCP
+    targetPort: 9901
+  - port: 9902
+    protocol: UDP
+    targetPort: 9902
+  - port: 9903
+    protocol: TCP
+    targetPort: 9903
+```
+
+You must also configure {{site.base_gateway}}'s [`proxy_listen`](/gateway/configuration/#proxy-listen) and [`stream_listen`] (/gateway/configuration/#stream-listen) configuration parameters in the container environment:
+
+```console
+KONG_PROXY_LISTEN="0.0.0.0:8000 reuseport backlog=16384, 0.0.0.0:8443 http2 ssl reuseport backlog=16384 http2"
+KONG_STREAM_LISTEN="0.0.0.0:9901 reuseport backlog=16384, 0.0.0.0:9902 reuseport backlog=16384 udp", 0.0.0.0:9903 reuseport backlog=16384 ssl"
+```
+
+Both the Service and `proxy_listen` configuration are managed via [the Helm chart](https://github.com/Kong/charts/tree/main/charts/kong) using the `proxy` configuration block.
+
+```yaml
+proxy:
+  http:
+    enabled: true
+    servicePort: 80
+    containerPort: 8000
+
+  tls:
+    enabled: true
+    servicePort: 443
+    containerPort: 8443
+
+  stream:
+    - containerPort: 9901
+      servicePort: 9901
+      protocol: TCP
+    - containerPort: 9902
+      servicePort: 9902
+      protocol: UDP
+    - containerPort: 9903
+      servicePort: 9903
+      protocol: TCP
+      parameters:
+        - "ssl"
+```
+
+Ports missing appropriate Kong-side configuration results in an error condition in the Gateway's status.
+
+```text
+message: no Kong listen with the requested protocol is configured for the requested port
+reason: PortUnavailable
+```
+
+### Listener compatibility and handling multiple Gateways
+
+Each {{ site.kic_product_name }} can be provided with a controller name. If no controller name is provided through the `--gateway-api-controller-name` field (or `CONTROLLER_GATEWAY_API_CONTROLLER_NAME` environment variable), the default `konghq.com/kic-gateway-controller` is used. 
+
+All the `GatewayClass`es referencing such a controller in the `controllerName` field are reconciled by the {{ site.kic_product_name }}. Similarly, all the `Gateway`s referencing a `GatewayClass` that specifies a matching `controllerName` are reconciled.
```

Comment:
```suggestion
Every `GatewayClass` referencing such a controller in the `controllerName` field is reconciled by the {{ site.kic_product_name }}. Similarly, every `Gateway` referencing a `GatewayClass` that specifies a matching `controllerName` is reconciled.
```
Rephrasing to avoid having weird plurals after codephrase formatting.